source/MaterialXRuntime/Private/Commands/PvtRemoveMetadataCmd.h

```diff
@@ -0,0 +1,35 @@
+//
+// TM & (c) 2020 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#ifndef MATERIALX_PVTREMOVEMETADATACMD_H
+#define MATERIALX_PVTREMOVEMETADATACMD_H
+
+#include <MaterialXRuntime/Private/PvtCommand.h>
+
+#include <MaterialXRuntime/RtValue.h>
+
+namespace MaterialX
+{
+
+class PvtRemoveMetadataCmd : public PvtCommand
+{
+public:
+    PvtRemoveMetadataCmd(RtObject& obj, const RtToken& name);
+
+    static PvtCommandPtr create(RtObject& obj, const RtToken& name);
+
+    void execute(RtCommandResult& result) override;
+    void undo(RtCommandResult& result) override;
+    void redo(RtCommandResult& result) override;
+
+private:
+    RtObject& _obj;
```

Comment:
This must be stored as a RtObject and not an RtObject&. RtObject is a shared pointer, and storing a reference (&) to a shared pointer will not increment the reference count. As a result this command class will not take any ownership of the object here so if all other references to it are released the object will be destroyed, even though this command might still be on the command queue.